src/results-output/results-doc.ts

```diff
@@ -236,85 +236,143 @@ export function appendCurrentTopLevelForm() {
     appendFormGrabbingSessionAndNS(true);
 }
 
+function lastLineIsEmpty(doc: vscode.TextDocument): boolean {
+  const lines = doc.lineCount;
+  const lastLine = doc.getText(new vscode.Range(lines - 1, 0, lines - 1, Infinity));
+  return lastLine === "";
+}
+
 let scrollToBottomSub: vscode.Disposable;
 export interface OnAppendedCallback {
     (insertLocation: vscode.Location, newPosition?: vscode.Location): any
 }
-let editQueue: [string, OnAppendedCallback][] = [];
-let applyingEdit = false;
-/* Because this function can be called several times asynchronously by the handling of incoming nrepl messages and those,
-   we should never await it, because that await could possibly not return until way later, after edits that came in from elsewhere
-   are also applied, causing it to wait for several edits after the one awaited. This is due to the recursion and edit queue, which help
-   apply edits one after another without issues.
-
-   If something must be done after a particular edit, use the onResultAppended callback. */
+
+type EditQueueItem = {
+  line: string ;
+  onAppended?: OnAppendedCallback;
+};
+type QueuedCallback = (item: EditQueueItem) => void;
+export class EditQueue {
```

Comment:
Would you mind refactoring this code to be more functional (similar to how it was, but with the new functionality you've applied)? I'd prefer we not use a class for this. I think classes tend to increase complexity and increase the likelihood of bugs and/or hard to maintain code.

I could help with this if you want.

Reply:
Actually, maybe hold off on that. I left a comment on the issue about a repro case: https://github.com/BetterThanTomorrow/calva/issues/1509#issuecomment-1052792339. I want to make sure we understand the cause better before we make changes.